google/cloud/bigtable/data/_metrics/handlers/opentelemetry.py

```diff
@@ -0,0 +1,136 @@
+# Copyright 2023 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from __future__ import annotations
+
+from uuid import uuid4
+
+from google.api.monitored_resource_pb2 import MonitoredResource  # type: ignore
+from google.cloud.bigtable import __version__ as bigtable_version
+from google.cloud.bigtable.data._metrics.handlers._base import MetricsHandler
+from google.cloud.bigtable.data._metrics.data_model import OperationType
+from google.cloud.bigtable.data._metrics.data_model import CompletedOperationMetric
+
+
+class OpenTelemetryMetricsHandler(MetricsHandler):
+    """
+    Maintains a set of OpenTelemetry metrics for the Bigtable client library,
+    and updates them with each completed operation and attempt.
+
+    The OpenTelemetry metrics that are tracked are as follows:
+      - operation_latencies: latency of each client method call, over all of it's attempts.
+      - first_response_latencies: latency of receiving the first row in a ReadRows operation.
+      - attempt_latencies: latency of each client attempt RPC.
+      - retry_count: Number of additional RPCs sent after the initial attempt.
+      - server_latencies: latency recorded on the server side for each attempt.
+      - connectivity_error_count: number of attempts that failed to reach Google's network.
+      - application_latencies: the time spent waiting for the application to process the next response.
+      - throttling_latencies: latency introduced by waiting when there are too many outstanding requests in a bulk operation.
+    """
+
+    def __init__(
+        self,
+        *,
+        project_id: str,
+        instance_id: str,
+        table_id: str,
+        app_profile_id: str | None,
+        client_uid: str | None = None,
+        **kwargs,
+    ):
+        super().__init__()
+        from opentelemetry import metrics
+
+        meter = metrics.get_meter(__name__)
+        self.op_latency = meter.create_histogram(
+            name="operation_latencies",
+            description="A distribution of the latency of each client method call, across all of it's RPC attempts",
+            unit="ms",
+        )
+        self.first_response_latency = meter.create_histogram(
+            name="first_response_latencies",
+            description="A distribution of the latency of receiving the first row in a ReadRows operation.",
+            unit="ms",
+        )
+        self.attempt_latency = meter.create_histogram(
```

Comment:
I think the histogram buckets need to be aligned with java

Reply:
My understanding was that the buckets are added as a later post-processing step, so I was going to look into that when I started looking at the exporter. But let me know if you want me to try to add it to this PR